spec/schedulers/TestScheduler-spec.ts

```diff
@@ -140,12 +140,31 @@ describe('TestScheduler', () => {
       expect(time).to.equal(50);
     });
 
+    it('should progress time with whitespace', () => {
+      const scheduler = new TestScheduler(null!);
+      const time = scheduler.createTime('     |');
+      //                                 -----|
+      expect(time).to.equal(50);
+    });
+
+    it('should progress time with mix of whitespace and dashes', () => {
+      const scheduler = new TestScheduler(null!);
+      const time = scheduler.createTime('  --|');
+      expect(time).to.equal(40);
+    });
+
     it('should throw if not given good marble input', () => {
       const scheduler = new TestScheduler(null!);
       expect(() => {
         scheduler.createTime('-a-b-#');
       }).to.throw();
     });
+
+    it('should ignore second stop', () => {
+      const scheduler = new TestScheduler(null!);
+      const time = scheduler.createTime('-----|--|');
+      expect(time).to.equal(50);
+    });
```

Comment:
issue: I don't think we should include a test like this. I mean, this isn't a valid marble diagram, so it should really throw. It just so happens that the `indexOf`-based implementation will pass this. It would also pass with `^#^#^|` - another completely invalid diagram. IMO, this could confuse someone reading the tests, so I think we'd be better off without it.

Reply:
I think I wanted to show that it will not throw if found another `!` sign like it throws here:
https://github.com/ReactiveX/rxjs/blob/adbe65e659bbf17f6ab20a9b30fcca0e4d76af9a/src/internal/testing/TestScheduler.ts#L246-L247
I agree they might confuse people, I will remove these tests.